### PR TITLE
Revive the fuzz testing framework

### PR DESCRIPTION
**Pull request by ionut-arm**
This commit updates the fuzz testing framework to adapt to recent
changes in the Parsec service code, allowing us to run the fuzzer within
the same container that we use for CI.

Signed-off-by: Ionut Mihalcea <ionut.mihalcea@arm.com>

### CODE REVIEW

File: fuzz/Cargo.toml [hunk 1]
```diff
@@ -20,7 +20,7 @@ lazy_static = "1.4.0"
 arbitrary = { version = "0.4.0", features = ["derive"] }
 
 [build-dependencies]
-parsec-client = "0.4.0"
+parsec-client = "0.12.0"
```
Comment (hug-dev) [inline fuzz/Cargo.toml:23]: time flies 🐦

**Review by hug-dev — APPROVED** [review]
Looks good!!
Could it be possible to add a very short fuzzing run (maybe only building?) to the per-PR CI so that we are sure that it still works overtime?

**Comment by ionut-arm** [comment]
Short test is now part of the CI

**Comment by hug-dev** [comment]
Perfect!

**Pull request merged**